src/main/java/com/aws/greengrass/shadowmanager/sync/SyncHandler.java

```diff
@@ -277,4 +297,87 @@ public void pushLocalDeleteSyncRequest(String thingName, String shadowName, byte
             overallSyncStrategy.putSyncRequest(new LocalDeleteSyncRequest(thingName, shadowName, deletePayload));
         }
     }
+
+    /**
+     * If add on interaction feature is enabled then adds the shadow to synchronization.
+     *
+     * @param thingName - The thing name to be added to synchronization
+     * @param shadowName - The shadow name to be added to synchronization
+     */
+    public void addShadowOnInteraction(String thingName, String shadowName) {
```

Comment:
I don't think `SyncHandler` should be responsible for checking and updating the configuration for this add on interaction feature. You can use the updater class to do that for you.